includes/admin.form.inc

```diff
@@ -6,287 +6,1073 @@
-    '#type' => 'textfield',
-    '#title' => t('Zoom per click'),
-    '#size' => 10,
-    '#element_validate' => array('element_validate_number'),
-    '#default_value' => $settings['zoomPerClick'],
-    '#description' => t('The factor by which images should zoom when clicked on.'),
-  );
-  // Zoom per scroll.
-  $form['islandora_openseadragon_settings']['zoomPerScroll'] = array(
-    '#type' => 'textfield',
-    '#title' => t('Zoom per scroll'),
-    '#size' => 10,
-    '#element_validate' => array('element_validate_number'),
-    '#default_value' => $settings['zoomPerScroll'],
-    '#description' => t('The factor by which images should zoom when scrolled over.'),
-  );
-  // Zoom per second.
-  $form['islandora_openseadragon_settings']['zoomPerSecond'] = array(
-    '#type' => 'textfield',
-    '#title' => t('Zoom per second'),
-    '#size' => 10,
-    '#element_validate' => array('element_validate_number'),
-    '#default_value' => $settings['zoomPerSecond'],
-    '#description' => t('The factor by which images should zoom every second when the zoom buttons are held down.'),
-  );
-  // Default zoom level.
-  $form['islandora_openseadragon_settings']['defaultZoomLevel'] = array(
-    '#type' => 'textfield',
-    '#title' => t('Default zoom level'),
-    '#size' => 10,
-    '#element_validate' => array('element_validate_number'),
-    '#default_value' => $settings['defaultZoomLevel'],
-    '#description' => t('The zoom level used by default for loaded OpenSeadragon viewers.'),
-  );
-
-  // Tile size.
-  $form['islandora_openseadragon_tile_size'] = array(
-    '#type' => 'textfield',
-    '#title' => t('Tile size'),
-    '#size' => 10,
-    '#element_validate' => array('element_validate_integer'),
-    '#default_value' => variable_get('islandora_openseadragon_tile_size', '256'),
-    '#description' => t('The size of image tiles used to compose the image as a whole, in pixels.'),
-  );
-  // Tile overlap.
-  $form['islandora_openseadragon_tile_overlap'] = array(
-    '#type' => 'textfield',
-    '#title' => t('Tile overlap'),
-    '#size' => 10,
-    '#element_validate' => array('element_validate_integer'),
-    '#default_value' => variable_get('islandora_openseadragon_tile_overlap', '0'),
-    '#description' => t('The number of pixels by which each tile will overlap adjacent ones.'),
-  );
-  $form['islandora_openseadragon_fit_to_aspect_ratio'] = array(
-    '#type' => 'checkbox',
-    '#title' => t('Constrain image to viewport'),
-    '#default_value' => variable_get('islandora_openseadragon_fit_to_aspect_ratio', FALSE),
+  $version = islandora_openseadragon_get_installed_version();
+  $djatoka_url = islandora_system_settings_form_default_value('islandora_openseadragon_djatoka_url', 'adore-djatoka/resolver', $form_state);
+  $iiif_url = islandora_system_settings_form_default_value('islandora_openseadragon_iiif_url', 'loris', $form_state);
+  $form = array(
+    'islandora_opensedragon_tilesource' => array(
```

Comment:
Also, this should make its way into the array of variables to delete upon module uninstallation.